test/index.js

```diff
@@ -451,4 +473,79 @@ describe('KeyringController', function () {
       );
     });
   });
+
+  describe('Manifest v3 changes', function () {
+    it('can still verify password', async function () {
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to verify the password
+      const result = await keyringController.verifyPassword(password);
+      expect(result).toBeTruthy();
+    });
+
+    it('can still unlock with password after being migrated and locked', async function () {
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to unlock the keychaing via old password
+      const result = await keyringController.submitPassword(password);
+      expect(result).toBeTruthy();
+
+      // Ensure the new vault value has separator and salt
+      const { vault, salt } = keyringController.store.getState();
+      expect(vault).toContain(':::');
+      expect(salt).toBeTruthy();
+    });
+
+    // TODO:  THIS SHOULD BE FAILING
+    it('does not unlock with incorrect encryption key', async function () {
+      await keyringController.submitPassword(password);
+
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to login with an incorrect encryption key
+      const encryptionKey = await keyringController.submitEncryptedKey(
+        'FAKE KEY',
+      );
+
+      expect(encryptionKey).rejects.toThrow();
+    });
+
+    it('unlocks with correct encryption key', async function () {
+      await keyringController.submitPassword(password);
+
+      const encryptionKey = keyringController.getEncryptedKey();
+
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to login with an incorrect encryption key
+      const key = await keyringController.submitEncryptedKey(encryptionKey);
+
+      expect(key).toBeTruthy();
```

Comment:
Is it in any way important (or relevant) that key is truthy?
Seems like this should assert  submitEncryptedKey() doesn't reject the promise